src/Handlers/AbstractHandler.php

```diff
@@ -265,7 +268,7 @@ protected function initializeAgents(): void
     // it's internal array.  then, we just call their initialize methods
     // in sequence.
     
-    if ($this->agentCollection instanceof AgentCollectionInterface) {
+    if (isset($this->agentCollection)) {

```

Comment:
the `agentCollection` property is always an instance of this Interface because we say it is when we declare it above.  what it may not be is set by the time we get here.  so, only if it is set will we run our foreach loop.